include/lsst/afw/geom/SphPoint.h

```diff
@@ -0,0 +1,319 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program. If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#ifndef LSST_AFW_GEOM_SPHPOINT_H_
+#define LSST_AFW_GEOM_SPHPOINT_H_
+
+#include <ostream>
+#include <utility>
+
+#include "lsst/afw/geom/Angle.h"
+#include "lsst/afw/geom/Point.h"
+
+namespace lsst {
+namespace afw {
+namespace geom {
+
+    /**
+     * @brief Point in an unspecified spherical coordinate system.
+     *
+     * This class represents a point on a sphere in the mathematical rather
+     * than the astronomical sense; it does not refer to any astronomical
+     * reference frame. Except where otherwise stated, all operations referring
+     * to two or more SphPoint objects assume all points are expressed in the
+     * same coordinate system.
+     *
+     * This class was not designed for inheritance, and must not be subclassed.
+     *
+     * @see @ref coord::Coord
+     */
+    // TODO: add final keyword after migrating to Pybind11
+    class SphPoint
+    {
+    public:
+        /**
+         * @brief Represent a specific pair of spherical coordinates.
+         *
+         * @param longitude The azimuth angle of the point.
+         * @param latitude The elevation angle of the point. Must correspond
+         *                 to an angle in [-&pi;/2, &pi;/2] radians.
+         *
+         * @throws InvalidParameterError Thrown if any argument is NaN or
+         *                               infinite.
+         * @throws OutOfRangeError Thrown if @c latitude is out of range.
+         *
+         * @par Exception Safety:
+         * The program state shall be unchanged in the event of an exception.
+         */
+        SphPoint(Angle const & longitude, Angle const & latitude);
+
+        /**
+         * @brief Represent a specific direction in 3-dimensional space.
+         *
+         * The adopted convention is that <1, 0, 0> corresponds to (0, 0), that
+         * <0, 1, 0> corresponds to (&pi;/2, 0), and that <0, 0, 1> corresponds
+         * to (0, &pi;/2).
+         *
+         * @param vector A position whose direction will be stored as a point.
+         *               Must not be the zero vector. Need not be normalized.
+         *
+         * @throws InvalidParameterError Thrown if any element of @c vector is
+         *                               NaN or infinite, or if @c vector is
+         *                               the zero vector.
+         *
+         * @par Exception Safety:
+         * The program state shall be unchanged in the event of an exception.
+         */
+        explicit SphPoint(Point3D const & vector);
+
+        /**
+         * @brief Creates a copy of a SphPoint.
```

Comment:
I personally prefer `SphPoint` to go with the `sphgeom` package. I don't think `SpherePoint` helps much, and `SphericalPoint` (the fully written out version) seems far longer than necessary.